test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java

```diff
@@ -531,6 +546,13 @@ public void onTimeout(TimeValue timeout) {
     }
 
     protected ActionFuture<CreateSnapshotResponse> startFullSnapshot(String repoName, String snapshotName, boolean partial) {
+        return startFullSnapshot(logger, repoName, snapshotName, partial);
+    }
+
+    public static ActionFuture<CreateSnapshotResponse> startFullSnapshot(Logger logger,
```

Comment:
I don't really like passing a logger around like this, I find it much easier to track down messages if the logger is just the class that logs the message even if it's a base class.

Reply:
It's more a preference than a requirement. Me, I'd have a static logger, I've never had a problem working out which test is running at any given time in the logs, but if you've a good reason for it then that's ok with me.

Reply:
Just want to weigh in that I also have a preference (but not a requirement) for static loggers.